### PR TITLE
Add kmtricks v1.4.0

### PR DESCRIPTION
**Pull request by tlemane**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by tlemane** [comment]
@BiocondaBot please add label

File: recipes/kmtricks/build.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                    -DCONDA_BUILD=ON \
+                    -DWITH_MODULES=ON \
+                    -DWITH_HOWDE=ON \
+                    -DKMER_LIST="${KMER_SIZE_LIST}" \
+                    -DWITH_SOCKS=ON
+cmake --build ./build
+
+cp -r ./bin/kmtricks $PREFIX/bin
+cp -r ./bin/kmtricks-socks $PREFIX/bin
+
```
Comment (bgruening) [inline recipes/kmtricks/build.sh:18]: Can you please add a comment what this is doing?

File: recipes/kmtricks/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make
+    - cmake
+
+  host:
+    - zlib
+
+  run:
```
Comment (bgruening) [inline recipes/kmtricks/meta.yaml:26]: this can be removed

**Review by bgruening — COMMENTED** [review]
Thanks @tlemane 
A few comments inline.

File: recipes/kmtricks/build.sh (hunk 1 above)
Comment (bgruening) [reply recipes/kmtricks/build.sh:27]: Sorry I meant the new complication and the third binary

Comment (tlemane) [reply recipes/kmtricks/build.sh:27]: kmtricks supports plugins through runtime dynamic linking. As this implies an overhead, plugin support is provided in a separate binary, called kmtricks**p**. I will add a comment in the script. Thanks for the review.

**Comment by tlemane** [comment]
@bgruening
I am finally giving up osx support for now. I have to fix some issues with dependencies. I will put it back in the next release.

**Review by shenwei356 — APPROVED** [review]

**Pull request merged**